compute/src/main/java/org/jclouds/compute/domain/internal/ArbitraryCpuRamTemplateBuilderImpl.java

```diff
@@ -60,24 +66,23 @@ protected Hardware findHardwareWithId(Set<? extends Hardware> hardwaresToSearch)
       } catch (NoSuchElementException ex) {
          if (isAutomaticId(hardwareId)) {
             AutomaticHardwareIdSpec spec = parseId(hardwareId);
-            return automaticHardwareForCpuAndRam(spec.getCores(), spec.getRam());
+            return automaticHardware(spec.getCores(), spec.getRam(), spec.getDisk());
          }
          else {
             throw ex;
          }
-        }
-    }
+      }
+   }
 
    protected Hardware resolveHardware(Set<? extends Hardware> hardwarel, final Iterable<? extends Image> images) {
       try {
          return super.resolveHardware(hardwarel, images);
       }
       catch (NoSuchElementException ex) {
-         if (super.minCores != 0 && super.minRam != 0) {
-            return automaticHardwareForCpuAndRam(minCores, minRam);
+         if (minCores <= 0 || minRam == 0 || minDisk < 0) {
+            throw new IllegalArgumentException("Please, set minCores, minRam and minDisk to positive values.");
```

Comment:
Keep the first part of the original message too:
`"No hardware profile matching the given criteria was found. If you want to use exact values, please set the minCores, minRam and minDisk to positive values"`